File: Tanishq Singhai/Find the third maximum element in an array.cpp
class Solution {
public:
    int thirdMax(vector<int>& nums) {
        // TC - O(N)
        // SC - O(1)
        long n = nums.size(), first = LONG_MIN, second = LONG_MIN, third = LONG_MIN;
        for (int i = 0; i < n; i++) {
            if (nums[i] == first || nums[i] == second || nums[i] == third)
                continue;
            if (nums[i] > first) {
                third = second;
                second = first;
                first = nums[i];
            } else if (nums[i] > second) {
                third = second;
                second = nums[i];
            } else if (nums[i] > third) {
                third = nums[i];
            }
        }
        return third == LONG_MIN ? first : third;
    }
};

File: Parth Mathur/Find the third maximum element in an array.java
class Solution {
public int thirdMax(int[] nums) {
Arrays.sort(nums);
int max1=-2147483648;
int max2=-2147483648;
int max3=-2147483648;
for(int i=0;i<nums.length;i++)
{
if(max1<nums[i])
{
max3=max2;
max2=max1;
max1=nums[i];
}
}
if(nums.length>=3 && max3==(-2147483648)&&(nums[0]|nums[1])>0)
{
max3=max1;
}
else if(nums.length==1)
{
max3=nums[0];
}
else if(nums.length==2)
{
max3=max1;
}
else if((max2|max3)==-2147483648)
max3=max1;
return max3;
}}
